### PR TITLE
Location, Post Form, Styling 

### PR DESCRIPTION
**Pull request by qihongw08**
### Changes

**Assets**
- Added styling to each individual components under assets folder 

**Services**
`LocationRequest.jsx` , `locationService.jsx` for location related logic 
`postService.jsx` for post related API calls 

**UX** 
- Modified `Login.jsx` when user is already logged in, auto direct to quotes page 
- Modifed `Quotes.jsx` to direct to login page if user is not logged in 

**`Posts.jsx` and `PostForm.jsx`**
- You can see all the posts in the Posts page
- You can post in the post form

### CODE REVIEW

File: src/components/Login.jsx [hunk 1]
```diff
@@ -1,54 +1,76 @@
...
+        await dispatch(isValidUser());
+      };
+      validateUser();
+    }, [dispatch])
 
-    const handleLogin = async (event, username, password) => {
-        event.preventDefault()
-        dispatch(initializeUser(username, password))
+    useEffect(() => {
+      console.log(user)
```
Comment (quntao-z) [inline src/components/Login.jsx:46]: remove

File: src/components/Login.jsx [hunk 2]
```diff
@@ -1,54 +1,76 @@
...
+      if (user.token) {
+        alert("welcome")
```
Comment (quntao-z) [inline src/components/Login.jsx:48]: do we still need this?

**Review by quntao-z — CHANGES_REQUESTED** [review]

File: src/common/BackendApi.js [hunk 3]
```diff
@@ -1,20 +1,53 @@
...
-      })
-      .catch(function (error) {
-        console.log(error);
-      });
-}
\ No newline at end of file
+const baseUrl = "/api/posts";
+
+// Retrieve the token from local storage
+const loggedUserJSON = window.localStorage.getItem("loggedUser");
```
Comment (quntao-z) [inline src/common/BackendApi.js:6]: Why don't we get token from user reducer here?

File: src/components/PostForm.jsx [hunk 4]
```diff
@@ -1,37 +1,65 @@
...
 const PostForm = () => {
-  const[rant, setRant] = useState('');
-  const[author, setAuthor] = useState('');
-
-  const handleRantChange = (e) => {
-    setRant(e.target.value);
-  }
+  const [rant, setRant] = useState('');
+  const [author, setAuthor] = useState('');
+  const token = "Bearer " + useSelector((state) => state.user.token);
```
Comment (quntao-z) [inline src/components/PostForm.jsx:11]: Can we handle this on the api side?

**Review by quntao-z — CHANGES_REQUESTED** [review]

File: src/common/BackendApi.js (hunk 3 above)
Comment (qihongw08) [reply src/common/BackendApi.js:6]: Backend Api is not used for now

File: src/reducer/userReducer.jsx [hunk 5]
```diff
@@ -34,9 +35,8 @@ export const initializeUser = (username, password) => {
         window.localStorage.setItem(
           'loggedUser', JSON.stringify(user)
         )
-        quoteService.setToken(user.token)
-        console.log(user.token)
-        dispatch(setUser(user))
+        quoteService.setToken(user.token);
```
Comment (quntao-z) [inline src/reducer/userReducer.jsx:38]: PostService set token here as well

**Review by quntao-z — CHANGES_REQUESTED** [review]

**Review by quntao-z — APPROVED** [review]

**Pull request merged**